From 9671d86e80a26b56b8c24f62a1c3210d3e5b5b7b Mon Sep 17 00:00:00 2001
From: Emma Foley <efoley@redhat.com>
Date: Tue, 11 Jan 2022 09:17:53 +0000
Subject: [PATCH] Include collectd.d and disable default loading

This patch will disable plugins loaded by default and will
apply config read from collectd.d
---
 src/collectd.conf.in | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/src/collectd.conf.in b/src/collectd.conf.in
index 562a55d9..028635d8 100644
--- a/src/collectd.conf.in
+++ b/src/collectd.conf.in
@@ -4,6 +4,10 @@
 # http://collectd.org/
 #
 
+# If you edit this file, it will not be updated during updates or upgrades.
+# It might be better to refer to it as an example only, and add your own
+# configuration snippets to new files under /etc/collectd.d.
+
 ##############################################################################
 # Global                                                                     #
 #----------------------------------------------------------------------------#
@@ -58,7 +61,7 @@
 # accessed.                                                                  #
 ##############################################################################
 
-@LOAD_PLUGIN_SYSLOG@LoadPlugin syslog
+#@LOAD_PLUGIN_SYSLOG@LoadPlugin syslog
 @LOAD_PLUGIN_LOGFILE@LoadPlugin logfile
 @LOAD_PLUGIN_LOG_LOGSTASH@LoadPlugin log_logstash
 
@@ -108,7 +111,7 @@
 #@BUILD_PLUGIN_CONNECTIVITY_TRUE@LoadPlugin connectivity
 #@BUILD_PLUGIN_CONNTRACK_TRUE@LoadPlugin conntrack
 #@BUILD_PLUGIN_CONTEXTSWITCH_TRUE@LoadPlugin contextswitch
-@BUILD_PLUGIN_CPU_TRUE@@BUILD_PLUGIN_CPU_TRUE@LoadPlugin cpu
+#@BUILD_PLUGIN_CPU_TRUE@@BUILD_PLUGIN_CPU_TRUE@LoadPlugin cpu
 #@BUILD_PLUGIN_CPUFREQ_TRUE@LoadPlugin cpufreq
 #@BUILD_PLUGIN_CPUSLEEP_TRUE@LoadPlugin cpusleep
 @LOAD_PLUGIN_CSV@LoadPlugin csv
@@ -140,7 +143,7 @@
 #@BUILD_PLUGIN_INFINIBAND_TRUE@LoadPlugin infiniband
 #@BUILD_PLUGIN_INTEL_PMU_TRUE@LoadPlugin intel_pmu
 #@BUILD_PLUGIN_INTEL_RDT_TRUE@LoadPlugin intel_rdt
-@BUILD_PLUGIN_INTERFACE_TRUE@@BUILD_PLUGIN_INTERFACE_TRUE@LoadPlugin interface
+#@BUILD_PLUGIN_INTERFACE_TRUE@@BUILD_PLUGIN_INTERFACE_TRUE@LoadPlugin interface
 #@BUILD_PLUGIN_IPC_TRUE@LoadPlugin ipc
 #@BUILD_PLUGIN_IPMI_TRUE@LoadPlugin ipmi
 #@BUILD_PLUGIN_IPSTATS_TRUE@LoadPlugin ipstats
@@ -148,7 +151,7 @@
 #@BUILD_PLUGIN_IPVS_TRUE@LoadPlugin ipvs
 #@BUILD_PLUGIN_IRQ_TRUE@LoadPlugin irq
 #@BUILD_PLUGIN_JAVA_TRUE@LoadPlugin java
-@BUILD_PLUGIN_LOAD_TRUE@@BUILD_PLUGIN_LOAD_TRUE@LoadPlugin load
+#@BUILD_PLUGIN_LOAD_TRUE@@BUILD_PLUGIN_LOAD_TRUE@LoadPlugin load
 #@BUILD_PLUGIN_LPAR_TRUE@LoadPlugin lpar
 #@BUILD_PLUGIN_LOGPARSER_TRUE@LoadPlugin logparser
 #@BUILD_PLUGIN_LUA_TRUE@LoadPlugin lua
@@ -159,7 +163,7 @@
 #@BUILD_PLUGIN_MDEVENTS_TRUE@LoadPlugin mdevents
 #@BUILD_PLUGIN_MEMCACHEC_TRUE@LoadPlugin memcachec
 #@BUILD_PLUGIN_MEMCACHED_TRUE@LoadPlugin memcached
-@BUILD_PLUGIN_MEMORY_TRUE@@BUILD_PLUGIN_MEMORY_TRUE@LoadPlugin memory
+#@BUILD_PLUGIN_MEMORY_TRUE@@BUILD_PLUGIN_MEMORY_TRUE@LoadPlugin memory
 #@BUILD_PLUGIN_MIC_TRUE@LoadPlugin mic
 #@BUILD_PLUGIN_MODBUS_TRUE@LoadPlugin modbus
 #@BUILD_PLUGIN_MQTT_TRUE@LoadPlugin mqtt
@@ -197,7 +201,7 @@
 #@BUILD_PLUGIN_REDIS_TRUE@LoadPlugin redis
 #@BUILD_PLUGIN_ROUTEROS_TRUE@LoadPlugin routeros
 #@BUILD_PLUGIN_RRDCACHED_TRUE@LoadPlugin rrdcached
-@LOAD_PLUGIN_RRDTOOL@LoadPlugin rrdtool
+#@LOAD_PLUGIN_RRDTOOL@LoadPlugin rrdtool
 #@BUILD_PLUGIN_SENSORS_TRUE@LoadPlugin sensors
 #@BUILD_PLUGIN_SERIAL_TRUE@LoadPlugin serial
 #@BUILD_PLUGIN_SIGROK_TRUE@LoadPlugin sigrok
@@ -2209,3 +2213,4 @@
 #    </Type>
 #  </Host>
 #</Plugin>
+Include "/etc/collectd.d"
-- 
1.8.3.1

